plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramActionbarContributor.java

```diff
@@ -26,15 +36,61 @@
 	@Override
 	public void init(IActionBars bars) {
 		super.init(bars);
+
 		// workaround for https://bugs.eclipse.org/bugs/show_bug.cgi?id=346648
 		bars.setGlobalActionHandler(GlobalActionId.SAVE, null);
 		bars.getToolBarManager().add(new DocumentationMenuAction());
-		
+
+		removeAppearanceButtonsInSimulationAndTestingPerspective(bars);
+
 		// remove 'arrange all' and 'arrange selection' actions
 		bars.getToolBarManager().remove(ActionIds.MENU_ARRANGE);
 		bars.getMenuManager().findMenuUsingPath(ActionIds.MENU_DIAGRAM).remove(ActionIds.MENU_ARRANGE);
 	}
 
+	private void removeAppearanceButtonsInSimulationAndTestingPerspective(IActionBars bars) {
+		IWorkbench wb = PlatformUI.getWorkbench();
+		IWorkbenchWindow win = wb.getActiveWorkbenchWindow();
+
+		win.addPerspectiveListener(new IPerspectiveListener() {
+
+			@Override
+			public void perspectiveActivated(IWorkbenchPage page, IPerspectiveDescriptor perspective) {
+				setToolbarButtons(bars, perspective);
+			}
+
+			@Override
+			public void perspectiveChanged(IWorkbenchPage page, IPerspectiveDescriptor perspective, String changeId) {
+				setToolbarButtons(bars, perspective);
+			}
+
+			private void setToolbarButtons(IActionBars bars, IPerspectiveDescriptor perspective) {
+				boolean isSimulationPers = ID_PERSPECTIVE_SCT_SIMULATION.equals(perspective.getId());
+				enableAppearanceButtons(bars.getToolBarManager(), !isSimulationPers);

```

Comment:
What about the SCT Testing perspective?
We need to clearify the requirements here, it is either

- show the buttons **only** in modeling perspective, or
- show the buttons **always but not in** simulation and testing perspective

currently you have implemented:

- show the buttons always but not in simulation perspective

Reply:
Is not in this Repo